### PR TITLE
Redo #143

### PR DESCRIPTION
**Pull request by Roxana-P**
This is a redo of PR #143 and alternative (?) to #146. I was asked to rebase and I hope this is what was the intention behind that request.

### CODE REVIEW

File: gustaf/io/meshio.py [hunk 1]
```diff
@@ -18,58 +19,96 @@
...
+                        meshes[i][0].BC[
+                            f"{meshes[j][0].whatami}-nodes"
+                        ] = np.unique(meshes[j][0].elements)
+
+    # Ensures backwards-compartibility
+    if return_only_one_mesh:
+        # Return highest-dimensional mesh
+        return_value = meshes[0][0]
+    else:
+        return_value = [mesh[0] for mesh in meshes]
```
Comment (j042) [inline gustaf/io/meshio.py:109]:
```suggestion
    return_value = [mesh[0] for mesh in meshes]
```
You can just return the whole thing without `return_only_one_mesh` option

File: gustaf/io/meshio.py [hunk 2]
```diff
@@ -18,58 +19,96 @@
...
+        # mesh.BC is only defined for volumes and faces
+        for i in range(len(meshes)):
+            # Indicator for boundary
+            if meshes[i][1] in [3, 2]:
+                for j in range(i, len(meshes)):
+                    # Considers only (d-1)-dimensional subspaces
+                    if meshes[j][1] == meshes[i][1] - 1:
+                        meshes[i][0].BC[
+                            f"{meshes[j][0].whatami}-nodes"
+                        ] = np.unique(meshes[j][0].elements)
```
Comment (j042) [inline gustaf/io/meshio.py:102]: Do you often have BC for edges or is this some sort of masks for faces/volumes?

File: gustaf/io/meshio.py [hunk 3]
```diff
@@ -18,58 +19,96 @@
...
+    meshes = [
+        [
+            meshio_mapping[key][0](
+                vertices=vertices, elements=meshio_mesh.cells_dict[key]
+            ),
+            meshio_mapping[key][1],
+        ]
+        for key in meshio_mapping.keys()
+        if key in meshio_mesh.cells_dict.keys()
+    ]
```
Comment (j042) [inline gustaf/io/meshio.py:79]:
```suggestion
    meshes = [
        [
            meshio_mapping[key][0](
                vertices=vertices, elements=meshio_mesh.cells_dict[key]
            ),
            meshio_mapping[key][1],
        ]
        for key in meshio_mapping.keys()
        if key in meshio_mesh.cells_dict.keys()
    ]
```
can you please just write a normal for loop here?

File: gustaf/io/meshio.py [hunk 4]
```diff
@@ -18,58 +19,96 @@
     # from meshio import Mesh as MeshioMesh
 
 
-def load(fname):
+def load(fname, set_boundary=False, return_only_one_mesh=True):
```
Comment (j042) [inline gustaf/io/meshio.py:22]:
```suggestion
def load(fname, set_boundary=True):
```
I think this is a nicer default

File: gustaf/io/meshio.py (hunk 3 above)
Comment (j042) [reply gustaf/io/meshio.py:79]: this is not resolved

**Comment by clemens-fricke** [comment]
Except for the Boundary information, this PR is superseded by #158.

**Pull request closed without merging**